docs/_docs/2_3_querying_data.md

```diff
@@ -136,6 +136,16 @@ The Spark Datasource API is a popular way of authoring Spark ETL pipelines. Hudi
 datasources work (e.g: `spark.read.parquet`). Both snapshot querying and incremental querying are supported here. Typically spark jobs require adding `--jars <path to jar>/hudi-spark-bundle_2.11-<hudi version>.jar` to classpath of drivers 
 and executors. Alternatively, hudi-spark-bundle can also fetched via the `--packages` options (e.g: `--packages org.apache.hudi:hudi-spark-bundle_2.11:0.5.3`).
 
+### Snapshot query {#spark-snap-query}
+This method can be used to retrieve the data table at the present point in time.
+
+```scala
+val hudiIncQueryDF = spark
+     .read()
+     .format("org.apache.hudi")
+     .option(DataSourceReadOptions.QUERY_TYPE_OPT_KEY(), DataSourceReadOptions.QUERY_TYPE_SNAPSHOT_OPT_VAL())
+     .load(tablePath + "/*") //Include "/*" at the end of the path if the table is partitioned
```

Comment:
when partition path is ${tablePath}/a/b/c, the load would be changed to `tablePath + "/*/*/*/*"` accordingly.

Reply:
You are correct, updating. Checking the actual partitioned directories seems like something that we should add into the hudi framework itself.

Reply:
Well, it would be one fewer, no? For each partition level so tablePath + "/*/*/*"

Reply:
no, you would refer to http://hudi.apache.org/docs/quick-start-guide.html#query-data

Reply:
This is a separate point from documentation, but ideally wouldn't it be better for Hudi to figure out which sub-directories need to be read for the partitions, instead of expecting to be passed the tree level of the partitions?